rust/lance/src/dataset/scanner.rs

```diff
@@ -632,7 +634,12 @@ impl Scanner {
     }
 
     /// Find k-nearest neighbor within the vector column.
-    pub fn nearest(&mut self, column: &str, q: &Float32Array, k: usize) -> Result<&mut Self> {
+    pub fn nearest<T: ArrowPrimitiveType>(
+        &mut self,
+        column: &str,
+        q: &PrimitiveArray<T>,
```

Comment:
this is a breaking change, code like this can't work since now:
```
.nearest(column, q.as_primitive(), k)
```
because the compiler can't infer the type